vllm/model_executor/models/qwen2.py

```diff
@@ -256,7 +256,14 @@ def forward(
         return hidden_states, residual
 
 
-@support_torch_compile
+@support_torch_compile(
+    dynamic_arg_dims={
+        "input_ids": 0,
+        # dim 1 for mrope in shape (3, seq_len), else dim 0 in shape (seq_len, )
+        "positions": lambda tensor: tensor.ndim - 1,
```

Comment:
does `-1` work here?

Reply:
you can do the conversion here:

https://github.com/vllm-project/vllm/blob/d06e824006d1ba4b92871347738ce1b89f658499/vllm/compilation/decorators.py#L177

iterate over the `dims` , and conver `-1` to `tensor.ndim - 1`